test/RToken.test.ts

```diff
@@ -1427,113 +1471,63 @@ describe(`RTokenP${IMPLEMENTATION} contract`, () => {
+      await Promise.all(
+        tokensAndAmounts.map(([token, amount]) =>
+          token.connect(addr1).approve(rToken.address, amount)
+        )
+      )
 
       // Issue rTokens
-      await rToken.connect(addr1).issue(issueAmount)
+      await rToken.connect(addr1)['issue(address,uint256)'](addr2.address, issueAmount)
 
-      expect(await token0.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn0))
-      expect(await token1.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn1))
-      expect(await token2.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn2))
-      expect(await token3.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn3))
+      // Check that that underlying was transferred
+      await Promise.all(
+        tokensAndAmounts.map(async ([token, amount]) => {
+          expect(await token.balanceOf(addr1.address)).to.equal(initialBal.sub(amount))
+        })
+      )
 
       // Check initial state
-      await expectUnprocessedIssuance(addr1.address, 0, {})
-      expect(await rToken.balanceOf(addr1.address)).to.equal(0)
+      await expectUnprocessedIssuance(addr2.address, 0, {})
+      expect(await rToken.balanceOf(addr2.address)).to.equal(0)
+
+      // Get initial balances
+      const initialRecipientBals = await Promise.all(tokens.map((t) => t.balanceOf(addr2.address)))
 
       // Cancel with issuer
-      await expect(rToken.connect(addr1).cancel(1, true))
+      await expect(rToken.connect(addr2).cancel(1, true))
         .to.emit(rToken, 'IssuancesCanceled')
-        .withArgs(addr1.address, 0, 1, issueAmount)
+        .withArgs(addr2.address, 0, 1, issueAmount)
 
-      expect(await rToken.balanceOf(addr1.address)).to.equal(0)
+      expect(await rToken.balanceOf(addr2.address)).to.equal(0)
 
-      // Check balances returned to user
-      expect(await token0.balanceOf(addr1.address)).to.equal(initialBal)
-      expect(await token1.balanceOf(addr1.address)).to.equal(initialBal)
-      expect(await token2.balanceOf(addr1.address)).to.equal(initialBal)
-      expect(await token3.balanceOf(addr1.address)).to.equal(initialBal)
+      // Check balances returned to the recipient, addr2
+      await Promise.all(
+        tokensAndAmounts.map(async ([token, amount], index) => {
+          const expectedBalance = amount.add(initialRecipientBals[index])
+          expect(await token.balanceOf(addr2.address)).to.equal(expectedBalance)
+        })
+      )
 
       // Check total asset value did not change
       expect(await facadeTest.callStatic.totalAssetValue(rToken.address)).to.equal(0)
 
       // We've cleared the queue so calls to cancel should revert
-      await expectProcessedIssuance(addr1.address, 0)
-      await expect(rToken.connect(addr1).cancel(1, true)).to.be.revertedWith('out of range')
+      await expectProcessedIssuance(addr2.address, 0)
+      await expect(rToken.connect(addr2).cancel(1, true)).to.be.revertedWith('out of range')
     })
```

Comment:
@tbrent addr1 issuing to addr2 and addr2 cancelling and receiving the underlying

Reply:
One more nit: can you include that when addr1 tries to cancel that the tx reverts and the cancellation does not occur? Other than this it looks ready to merge to me, but hold off until I get `pricing` into `master` later today.